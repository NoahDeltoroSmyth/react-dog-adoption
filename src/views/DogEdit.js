import React, { useState, useEffect } from 'react';
import DogForm from '../component/DogForm';
import { getDogsById, updateDog } from '../services/dogRoute';
import { useParams, useHistory } from 'react-router-dom';

export default function DogEdit() {
  const [name, setName] = useState('');
  const [breed, setBreed] = useState('');
  const [age, setAge] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getDogsById(params.id);
      setName(resp.name);
      setBreed(resp.breed);
      setAge(resp.age);
      setBio(resp.bio);
      setImage(resp.image);
    };
    fetchData();
  }, [params.id]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    const { status } = await updateDog(params.id, name, breed, age, bio, image);
    if (status >= 400) {
      alert('Status error');
    } else {
      alert('Successfully edited');
      history.push('/dogs');
    }
  };

  return (
    <div>
      <DogForm
        name={name}
        setName={setName}
        breed={breed}
        setBreed={setBreed}
        age={age}
        setAge={setAge}
        bio={bio}
        setBio={setBio}
        image={image}
        setImage={setImage}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
