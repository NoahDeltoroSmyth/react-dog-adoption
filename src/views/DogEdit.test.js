import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogEdit from './DogEdit';

it('should render a dogs info in the form', async () => {
  //render
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/21/edit']}>
      <Route path="/dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );
  //wait for load
  await screen.findByDisplayValue('Anna');
  //take screen shot
  expect(container).toMatchSnapshot();
});
