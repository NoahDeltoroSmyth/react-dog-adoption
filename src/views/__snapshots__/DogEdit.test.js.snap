// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a dogs info in the form 1`] = `
<div>
  <div>
    <div
      class="form"
    >
      <form>
        <h1>
          Dog Information
        </h1>
        <div
          class="form-control"
        >
          <label>
            Name:
          </label>
          <input
            placeholder="Name"
            type="text"
            value="Anna"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Breed:
          </label>
          <input
            placeholder="Breed"
            type="text"
            value="Longhair lab/shepherd mix"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Age:
          </label>
          <input
            placeholder="Age"
            type="text"
            value="23"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Bio:
          </label>
          <input
            placeholder="Bio"
            type="text"
            value="Always puts smiles on other's faces and enjoys the better things in life"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image:
          </label>
          <input
            placeholder="Image"
            type="text"
            value="https://placedog.net/1000/563?id=155"
          />
        </div>
        <div>
          <button
            class="button"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
